src/vcf/store_parse_policy.cpp

```diff
@@ -84,25 +84,22 @@ namespace ebi
     {
         // Put together m_line_typeid and m_grouped_tokens in a single MetaEntry object
         // Add MetaEntry to Source
-        try {
-            if (m_line_typeid == "") { // Plain value
-                state.add_meta(MetaEntry{m_grouped_tokens[0]});
 
-            } else if (m_grouped_tokens.size() == 1) { // TypeID=value
-                state.add_meta(MetaEntry{m_line_typeid, m_grouped_tokens[0]});
+        if (m_line_typeid == "") { // Plain value
+            state.add_meta(MetaEntry{state.n_lines, m_grouped_tokens[0]});
 
-            } else if (m_grouped_tokens.size() % 2 == 0) { // TypeID=<Key-value pairs>
-                auto key_values = std::map<std::string, std::string>{};
-                for (size_t i = 0; i < m_grouped_tokens.size(); i += 2) {
-                    key_values[m_grouped_tokens[i]] = m_grouped_tokens[i+1];
-                }
-                state.add_meta(MetaEntry{m_line_typeid, key_values});
+        } else if (m_grouped_tokens.size() == 1) { // TypeID=value
+            state.add_meta(MetaEntry{state.n_lines, m_line_typeid, m_grouped_tokens[0]});
 
-            } else {
-                // TODO Throw exception
+        } else if (m_grouped_tokens.size() % 2 == 0) { // TypeID=<Key-value pairs>
+            auto key_values = std::map<std::string, std::string>{};
+            for (size_t i = 0; i < m_grouped_tokens.size(); i += 2) {
+                key_values[m_grouped_tokens[i]] = m_grouped_tokens[i+1];
             }
-        } catch (std::invalid_argument ex) {
-            throw ParsingError(ex.what());
+            state.add_meta(MetaEntry{state.n_lines, m_line_typeid, key_values});
+
+        } else {
+            // TODO Throw exception
```

Comment:
Now that we actually have those exceptions hierarchy, could a `MetaSectionError` be thrown?